brave/brave-core
# Customize UA strings

Brave/+ chromium major version string(ex Brave/74) is appended to chromium ua.
To test with original UA, --disable-override-ua switch is introduced.

As we use custom UA, UA whitelist and referral header are not used anymore.

Fix brave/brave-browser#1052

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on

## Test Plan:
`yarn test brave_browser_tests --filter=BraveContentBrowserClientTest.OverrideUATest`
`yarn test brave_browser_tests --filter=BraveContentBrowserClientWithoutUAOverride.DisableOverrideUATest`

Manual test 1.
* Launch browser and open inspector
* Check "Brave/XX" is appended to ua by checking `navigator.userAgent`

Manual test 2.
* Launch browser with `--disable-override-ua` switch
* Check "Brave/XX" is *not* appended to ua by checking `navigator.userAgent`

Manual test 3.
* Launch browser and loading cheddar.com
* Check `x-brave-partner` is not used.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
